ceph/ceph_admin/bootstrap.py

```diff
@@ -401,10 +401,26 @@ def bootstrap(self: CephAdmProtocol, config: Dict):
             public_nws = ",".join(filter(lambda x: x, list(set(public_nws.split(",")))))
 
         if public_nws:
-            self.shell(args=["ceph", "config", "set", "mon public_network", public_nws])
+            # Refer:
+            # - https://bugzilla.redhat.com/show_bug.cgi?id=2314438
+            # - https://bugzilla.redhat.com/show_bug.cgi?id=2156919
+            # - https://access.redhat.com/solutions/7088483
+            # Todo: update public network config level to global for 6.x and 7.x when changes are in
+            # 6.x tracker: https://bugzilla.redhat.com/show_bug.cgi?id=2314604
+            # 7.x tracker: https://bugzilla.redhat.com/show_bug.cgi?id=2314606
+            config_level = "global" if rhbuild.split(".")[0] >= "8" else "mon"
+            self.shell(
+                args=[
+                    "ceph",
+                    "config",
+                    "set",
+                    f"{config_level} public_network",
+                    public_nws,
+                ]
+            )
         if cluster_nws:
             self.shell(
-                args=["ceph", "config", "set", "mon cluster_network", cluster_nws]
+                args=["ceph", "config", "set", "global cluster_network", cluster_nws]
```

Comment:
I believe we would need `config_level` variable here.

IMHO, we could

```
config_level = "global" if rhbuild.split(".")[0] >= "8" else "mon"
if public_nws:
  ...
```

Reply:
`cluster network` is always at `global level`, it should not be set at only `mon` level.
@pdhiran Please confirm the above.

Reply:
Yes, cluster_network is always set at global level, and mon level would not work. ( configs from mon level aren't propagated at OSD level. Whereas Configs at global level will be propagated at all daemon levels )


IBM doc link : https://www.ibm.com/docs/en/storage-ceph/7?topic=configuration-configuring-private-network

Reply:
@pdhiran @harshkumarRH thanks for the confirmation.